components/AddContentProviderForm/AddContentProviderForm.tsx

```diff
@@ -0,0 +1,64 @@
+import type { ChangeEventHandler, FormEventHandler } from 'react';
+import { useCallback, useState } from 'react';
+
+import { Button } from '../Button/Button';
+
+import styles from './addContentProviderForm.module.scss';
+import { useFormReducer } from './utils';
+
+export const AddContentProviderForm = () => {
+  const [contentURL, setContentURL] = useState('');
+  const [email, setEmail] = useState('');
+  const { setIsError, setIsLoading, setIsSuccess } = useFormReducer();
+
+  const handleContentURLChange: ChangeEventHandler<HTMLInputElement> = useCallback((e) => {
+    setContentURL(e.currentTarget.value);
+  }, []);
+
+  const handleEmailAddressChange: ChangeEventHandler<HTMLInputElement> = useCallback((e) => {
+    setEmail(e.currentTarget.value);
+  }, []);
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = async (e) => {
+    e.preventDefault();
+    setIsLoading();
+    try {
+      const response = await fetch('https://www.polskifrontend.pl/api/content-provider', {
```

Comment:
Nie wiem czy to się gdzieś powtórzy, ale może warto byłoby to wyrzucić jako oddzielna funkcja?